packages/client/tests/functional/too-many-bind-variables/tests.ts

```diff
@@ -0,0 +1,93 @@
+import testMatrix from './_matrix'
+// @ts-ignore
+import type * as imports from './node_modules/@prisma/client'
+
+declare let prisma: imports.PrismaClient
+
+testMatrix.setupTestSuite(
+  ({ provider, providerFlavor }) => {
+    const n = 32768
+
+    async function generatedIds(n: number) {
+      // ["1","2",...,"n"]
+      const ids = Array.from({ length: n }, (_, i) => i + 1).map((id) => `${id}`)
+      const idsAsString = ids.map((id) => ({ id, field: '' }))
+
+      if (provider === 'sqlite') {
+        // sqlite doesn't support `createMany` yet!
+        // See: https://github.com/prisma/prisma/issues/10710
+
+        const promises = ids.map((id) =>
+          prisma.model.create({
+            data: { id, field: '' },
+          }),
+        )
+        await Promise.all(promises)
+      } else {
+        await prisma.model.createMany({
+          data: idsAsString,
+        })
+      }
+
+      return ids
+    }
+
+    function selectWithIds(ids: string[]) {
+      return prisma.model.findMany({
+        where: {
+          OR: [
+            {
+              id: { in: ids },
+            },
+            {
+              id: { in: ids },
+            },
+          ],
+        },
+      })
+    }
+
+    testIf(providerFlavor === undefined)(
+      'Using Rust drivers with massive filter list results in "too many bind variables" error',
+      async () => {
+        const ids = await generatedIds(n)
+
+        try {
+          await selectWithIds(ids)
+
+          // unreachable
+          expect(true).toBe(false)
+        } catch (error) {
+          const e = error as Error
+
+          expect(e.message).toContain('Assertion violation on the database')
+          expect(e.message).toContain('too many bind variables in prepared statement')
```

Comment:
Shouldn't these work as the engine should take care of chunking to split the queries into multiple?

Reply:
That was also my expectation, same for this user in https://github.com/prisma/prisma/issues/21648#issuecomment-1783838900
Because we fixed something that looks similar in https://github.com/prisma/prisma/issues/8832#issuecomment-1260533913

Reply:
This is indeed a bug, the Postgres extended protocol uses a short to identify the number of query parameters, but we know that at the engine level, because we know wether rust drivers will run. We could fix that in this PR, but I don't think it's in the scope of this issue, but a correctness one, as it's not a bug in driver adapters code.

Reply:
Actually the investigation subject of https://github.com/prisma/team-orm/issues/394 is done without the need to merge this PR and the outcome should be a correctness issue to fix the binding problem when using rust drivers.

Reply:
We have had previous indications of Driver Adapters not using the query chunking logic defined in the Prisma engines.
Keeping [this](https://github.com/prisma/team-orm/issues/394#issuecomment-1787218882) in mind, I've answered in a top-level comment for future visibility. See: https://github.com/prisma/prisma/pull/21679#issuecomment-1787315586.